sdk/lib/drivers/sound/mmixer/controls.c

```diff
@@ -181,53 +182,44 @@ MMixerAddMixerControl(
 
         if (Status == MM_STATUS_SUCCESS)
         {
-            LPMIXERVOLUME_DATA VolumeData;
-            ULONG Steps, MaxRange, Index;
-            LONG Value;
-
             Members = (PKSPROPERTY_MEMBERSHEADER)(Desc + 1);
             Range = (PKSPROPERTY_STEPPING_LONG)(Members + 1);
 
             DPRINT("NodeIndex %u Range Min %d Max %d Steps %x UMin %x UMax %x\n", NodeIndex, Range->Bounds.SignedMinimum, Range->Bounds.SignedMaximum, Range->SteppingDelta, Range->Bounds.UnsignedMinimum, Range->Bounds.UnsignedMaximum);
 
-            MaxRange = Range->Bounds.UnsignedMaximum - Range->Bounds.UnsignedMinimum;
+            VolumeData = (LPMIXERVOLUME_DATA)MixerContext->Alloc(sizeof(*VolumeData));
+            if (!VolumeData)
+                return MM_STATUS_NO_MEMORY;
```

Comment:
It would appear you can move this allocation + check just before the `if (Status == MM_STATUS_SUCCESS)` , or even before the `MixerContext->Control()` call, and remove it in the else-block, since, in either case (whether the `MixerContext->Control()` call succeeds or not), you are allocating that volume data to use it afterwards.